blogs/13_bfcl_v3_multi_turn.html

```diff
@@ -748,73 +751,59 @@ <h4>5. API Code Validation (🧑‍💻+ 💻 )</h4>
-
-          <h4>1. Differences in Inference Patterns Between FC and Prompting Models</h4>
-          <p>For FC models, the output is generated in JSON format, where each function call is structured with clear
-            parameters and ready for direct execution by the API backend.</p>
-          <p>In contrast, prompting models generate outputs as function call strings. For instance, the model may
-            output a function call in the following format:</p>
-          <pre><code>"[write_to_file(filename='log.txt', content='hello'), close_file(filename='log.txt')]".</code></pre>
-          <p>These outputs require parsing before they can be executed. The string format is first parsed into
-            individual function calls, and each function call is then executed in the backend as if it were a normal
-            function-calling API.</p>
-          <p>The key distinction is that FC models produce structured, executable JSON directly, while prompting
-            models require post-processing to interpret and extract the function calls from the string format.</p>
-
-          <h4>2. Handling Different Multi-Turn Function Call Patterns</h4>
-          <p>Multi-turn function calling can present a variety of challenges in inference, particularly when it comes
-            to managing the flow of data and function results across multiple steps. In <i>BFCL V3 • Multi-Turn & Multi-Step</i>, our model handlers
-            are designed to handle different function call patterns—simple, parallel, and nested—across multiple
-            rounds of interaction. The distinction between these call patterns is explained in the previous section.
+          <h3 id="inference">Model Inference & Execution</h3>
+          <ol>
+            <li>
+              Each entry comes with its own <code>initial_config</code>, which is used to initialize the API backend state. For example, a file system backend might start with a set of pre-existing files, and a messaging API might start with a specific message inbox history. The <code>initial_config</code> is loaded into the API backend at the beginning of each test entry, but it is not provided to the model directly, as the model can call exploration functions to figure out the state of the system and retrieve the necessary information.
+            </li>
+            <li>
+              At the beginning of each turn, a user query is provided to the model, along with the tools available for the task. A <code>step_counter</code> is also initialized to track the number of steps used in this turn; it resets to 0 at the beginning of each turn.
+            </li>
+            <li>
+              If the model makes any function calls (i.e., <code>decode_exec</code> in the model handler returns a non-empty list), the function calls are executed in the API backend (in order) and their execution results are provided back to the model. The current turn continues.
+            </li>
+            <li>
```

Comment:
Can we have an explicit subheading here to mark when does a `step end`, and when does a `turn end`?

Reply:
It's still not very clear to me we are communicating this properly.

Reply:
Can we also provide here or at the end, why we made this design decision? What are the pros and what are the cons? (this ==> what we consider is an end of turn and what we consider is an end of step).